### PR TITLE
build: fix compilation issues in optimize

### PR DESCRIPTION
**Pull request by buccarel**
## Description

This PR fixes compilation issues detected in Optimize after the merge. In particular:
* the experimental Lombok annotation `@FieldNameConstants` was not being compiled correctly, so it is being removed in favor of its corresponding vanilla Java
* the `@Builder` annotation in one class was not being compiled, so it is being removed in favor of its corresponding vanilla Java
* the docstrings in `IgnoreDuringScan.java` had syntax issues that are being corrected

The command that I used to do the compilation is the following:
```bash
$ mvn -B -D skipTests -D skipChecks compile generate-sources source:jar javadoc:jar -P skipFrontendBuild
```

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

* https://github.com/camunda/camunda-optimize/issues/13962

### CODE REVIEW

File: optimize/backend/src/main/java/io/camunda/optimize/IgnoreDuringScan.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
  * @ComponentScan(excludeFilters = @ComponentScan.Filter(IgnoreDuringScan.class))
  * @Configuration
  * public class Main {
+ * }
```
Comment (buccarel) [inline optimize/backend/src/main/java/io/camunda/optimize/IgnoreDuringScan.java:21]: This was giving compilation error: even though it's in a comment block, it didn't close a docstrings tag well

File: optimize/backend/src/main/java/io/camunda/optimize/dto/optimize/importing/DecisionInstanceDto.java [hunk 2]
```diff
@@ -43,4 +41,25 @@ public class DecisionInstanceDto implements OptimizeDto {
   private Set<String> matchedRules = new HashSet<>();
   private String engine;
   private String tenantId;
+
+  public static final class Fields {
```
Comment (PHWaechtler) [inline optimize/backend/src/main/java/io/camunda/optimize/dto/optimize/importing/DecisionInstanceDto.java:45]: ❓ How come in some classes we're using a class like this one while in others (eg IdentityLinkLogEntryDto ) we're using an enum?

**Review by HeleneW-dot — APPROVED** [review]
Lgtm, but it would be nice to have a consistent implementation we can share across all classes, ie either always go for the enum or always go for the static class. Can we adjust this or is there a reason we need one or the other in separate classes?

**Review by matthewBearCamunda — COMMENTED** [review]
In general this LGTM. I didn't look through all of the classes, but they all follow the same pattern

One question though, why put the variables in a `Fields` inner class each time? Why not just have the variables in the regular class?

File: optimize/backend/src/main/java/io/camunda/optimize/dto/optimize/importing/DecisionInstanceDto.java (hunk 2 above)
Comment (buccarel) [reply optimize/backend/src/main/java/io/camunda/optimize/dto/optimize/importing/DecisionInstanceDto.java:45]: Also replying to @matthewBearCamunda 

Well this comes from how the annotation was used before. In some cases, it had the `asEnum = true` argument, in other cases it didn't have it. The 2 cases have a different vanilla Java code, so the results are different.

I don't know why it was this way originally. CC'ing @RomanJRW to see if he happens to have more context about this

**Comment by buccarel** [comment]
> In general this LGTM. I didn't look through all of the classes, but they all follow the same pattern
> 
> One question though, why put the variables in a `Fields` inner class each time? Why not just have the variables in the regular class?

@matthewBearCamunda that's how the annotation works. When you annotate a class, the class `Fields` is created. I didn't remove it now after de-lombok'ing because then I would need to change all of the consumers. Also, de-lombok'ing into a different format is not possible with automatic tools like the one I used

**Pull request merged**